### DISCUSSION TITLE
Latest Airflow 2.x version with old import naming schema?

### DISCUSSION THREAD

**Discussion by Cookie12321** [apache/airflow#28413] [General] [answered]
I've searched the Airflow release notes all the way from 2.0.0 to the present version, but I couldn't find a satisfactory answer.

Airflow 2 used to allow the old import naming schema: 
- `from airflow.operators.sensors import BaseSensorOperator`

And here's the current import naming schema:
- `from airflow.sensors import BaseSensorOperator`

According to this [stackoverflow comment](https://stackoverflow.com/a/66300001), Airflow 2 stopped supporting the backwards compatibility of the old import naming schema and now only supports the new (current) way. 

Which Airflow 2.x version ended up making this change permanent?

**Comment by potiuk** [comment, marked as answer]
2.0.0.

I am not sure how you searched for it, because it is explicitly mentioned in the release notes (I found it by Ctrl+F "operators.sensors") 

https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-0-0-2020-12-18

> Changes to sensor imports
> ~~~~~~~~~~~~~~~~~~~~~~~~~

> Sensors are now accessible via ``airflow.sensors`` and no longer via ``airflow.operators.sensors``.

> For example: ``from airflow.operators.sensors import BaseSensorOperator``
becomes ``from airflow.sensors.base import BaseSensorOperator``

**Reply by Cookie12321** [reply]
Thank you. I remember seeing another text saying that the import schema name change would be backwards compatible, and I confused what that meant.

Thanks for clearing it up.

**Reply by potiuk** [reply]
There were other backwards compatible imports. But not this one.
